infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java

```diff
@@ -158,7 +159,9 @@ public void createOrReplace(PersonalAccessToken personalAccessToken)
                               scmUrl.getProtocol(),
                               personalAccessToken.getScmTokenName().startsWith(OAUTH_2_PREFIX)
                                   ? "oauth2"
-                                  : personalAccessToken.getScmUserName(),
+                                  : isNullOrEmpty(personalAccessToken.getScmOrganization())
+                                      ? personalAccessToken.getScmUserName()
+                                      : personalAccessToken.getScmOrganization(),
```

Comment:
I haven't tested this yet but this appears to apply to all platforms (e.g. GitHub, etc.). Is this intentional? I'm generally a little confused on what works for all platforms -- e.g. on GitHub my username is `amisevsk` and (I guess) my organization name is `amisevsk`, even though I'm not an organization?

Reply:
According to the [PAT documentation](https://www.eclipse.org/che/docs/stable/end-user-guide/using-a-git-provider-access-token/) we use the `che.eclipse.org/scm-organization` annotation only for the Azure DevOps. For GitHub and others except Azure, `scmUsername` is used so the flow is the same.

Reply:
can we extract this logic to a separate method and add javadoc for that?

Reply:
done